src/TileChartMap.jsx

```diff
@@ -74,6 +75,43 @@ export class TileChartMap extends React.Component {
   }
 }
 
+/**
+ * helper function to get display value of tile based on selected dropdown.
+ * @param {object} props contains data and normalization
+ * @returns {object} data provided to tile map
+ */
+function updateData( props ) {
+  const { data, dataNormalization } = props
+  const showDefault = dataNormalization === GEO_NORM_NONE
+  const res = data[0].map( o => ( {
+    ...o,
+    displayValue: getDisplayValue( o, showDefault ),
+    perCapita: getPerCapita( o )
+  } ) )
+
+  return [ res ]
+}
+
+/**
+ * helper function to switch between the value and per capita values
+ * @param {object} stateObj contains value
+ * @param {boolean} showDefault whether default value should be shown or perCap
+ * @returns {number} value
+ */
+function getDisplayValue( stateObj, showDefault ) {
+  return showDefault ? stateObj.value : getPerCapita( stateObj )
+}
+
+/**
+ * helper function to calculate percapita value
+ * @param {object} stateObj a state containing abbr and value
+ * @returns {string} the per capita value
+ */
+function getPerCapita( stateObj ) {
+  const pop = STATE_DATA.find( o => o.abbr === stateObj.name ).population
```

Comment:
Any advantage to precomputing this on page load after render so it's available on click, rather than doing the read each time, or is that overoptimization for this?

Reply:
Whoops. you're right. I updated the logic to have it in the reducer.